### PR TITLE
[MISC] Updating issue templates

### PR DESCRIPTION
**Pull request by deusebio**
Introducing a new template for issues, based on [this](https://github.com/juju/terraform-provider-juju/tree/main/.github/ISSUE_TEMPLATE). 

The new template introduces different templates for Bugs and Feature Requests. The required information for bugs are mostly still there but on dedicated text area. 

I'd like to get a review from other teams such that we can hopefully stardardise this across the board.

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+      label: Juju client version
+      description: Version of the Juju client. Run `juju version` to obtain it.
+      placeholder: 3.1.7
+    validations:
+      required: true
+
+  - type: input
+    id: Juju-Version
+    attributes:
+      label: Juju client version
```
Comment (taurus-forever) [inline .github/ISSUE_TEMPLATE/BugReport.yml:43]: Juju controller version?

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      placeholder: 3.1.7
+    validations:
+      required: true
+
+  - type: input
+    id: Charm-Revision
+    attributes:
+      label: Charm revision version
+      description: What version of the Juju agent are you running? Run `juju status` to obtain it.
+      placeholder: 3.1.7
```
Comment (taurus-forever) [inline .github/ISSUE_TEMPLATE/BugReport.yml:54]: 42 ?
Charm revsisions are integers and not Semantic.

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+      description: What version of the Juju client are you running? Run `juju version` to obtain it.
+      placeholder: 3.1.7
+    validations:
+      required: true
+
+  - type: input
+    id: Charm-Revision
+    attributes:
+      label: Charm revision version
+      description: What version of the Juju agent are you running? Run `juju status` to obtain it.
```
Comment (taurus-forever) [inline .github/ISSUE_TEMPLATE/BugReport.yml:53]: Wrong comment?

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+      description: Version of the Juju client. Run `juju version` to obtain it.
+      placeholder: 3.1.7
+    validations:
+      required: true
+
+  - type: input
+    id: Juju-Version
+    attributes:
+      label: Juju client version
+      description: What version of the Juju client are you running? Run `juju version` to obtain it.
```
Comment (taurus-forever) [inline .github/ISSUE_TEMPLATE/BugReport.yml:44]: Need update?

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+      placeholder: 3.1.7
+    validations:
+      required: true
+
+  - type: input
+    id: Substrate-Specifics
+    attributes:
+      label: Substrate version or specification
+      description: Provide information about the substrate where you are running. For instance, on LXD use `lxd version`.
+      placeholder: "LXD: 5.0.3"
```
Comment (taurus-forever) [inline .github/ISSUE_TEMPLATE/BugReport.yml:63]: LXD: 5.0.3
MicropK8s 1.29.0-strict (revision 12345)
GKE: 1.29.1

?

File: .github/ISSUE_TEMPLATE/BugReport.yml [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+      description: "Please add the debug output you see when running the 
+      commands above. Consider also making a gist containing output of the `crash.log` too."
+      render: shell
+
+  - type: textarea
+    id: Info-Notes
+    attributes:
+      label: "Notes & References"
+      description: "Please add relevant notes, links to mattermost chats, 
+      other related issues/PRs, anything to help diagnose and solve the issue."
```
Comment (taurus-forever) [inline .github/ISSUE_TEMPLATE/BugReport.yml:94]: Do we need "Expected results" from reported?

My personal mandatory list:
* STR (steps-to-reproduce)
* Actual Results
* Expected Results
* Notes (version, logs, extra info ....)
Like we had earlier :-)

**Review by taurus-forever — COMMENTED** [review]
LGTM, but it is a holly war like spaces VS tabs. I do not mind to use Canonical standard template... sign one with Maksim? :-)

P.S. At the moment all our repos uses identical template and I like this, if you are planning update, we should update all repos and not Kafka. Otherwise LGTM.

P.P.S. IMHO templates must sit [here](https://github.com/canonical/data-platform) and be linked to all repos.

File: .github/ISSUE_TEMPLATE/BugReport.yml (hunk 1 above)
Comment (deusebio) [reply .github/ISSUE_TEMPLATE/BugReport.yml:43]: Sorry. I messed up a bit the template on this item and the next one

File: .github/ISSUE_TEMPLATE/BugReport.yml (hunk 2 above)
Comment (deusebio) [reply .github/ISSUE_TEMPLATE/BugReport.yml:54]: Yes, I messed up a bit this item and the one above.

File: .github/ISSUE_TEMPLATE/BugReport.yml (hunk 6 above)
Comment (deusebio) [reply .github/ISSUE_TEMPLATE/BugReport.yml:94]: Yes, just I'm not sure Actual Results and Expected Results would always apply so I have created a required "Description" but optional behaviour forms

**Comment by deusebio** [comment]
This will be superseded by another PR by @shayancanonical

**Pull request closed without merging**